lib/filterx/object-list.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2023 Balazs Scheidler <balazs.scheidler@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+#include "object-list.h"
+#include "object-string.h"
+#include "scanner/list-scanner/list-scanner.h"
+#include "str-repr/encode.h"
+
+#include <json-c/json.h>
+
+typedef struct _FilterXList
+{
+  FilterXObject super;
+  GPtrArray *objects;
+} FilterXList;
+
+static gboolean
+_truthy(FilterXObject *s)
+{
+  return TRUE;
+}
+
+static gboolean
+_marshal(FilterXObject *s, GString *repr, LogMessageValueType *t)
+{
+  FilterXList *self = (FilterXList *) s;
+
+  for (gint i = 0; i < self->objects->len; i++)
+    {
+      FilterXObject *elem = g_ptr_array_index(self->objects, i);
+
+      if (!filterx_object_is_type(elem, &FILTERX_TYPE_NAME(string)))
+        return FALSE;
+
+      if (repr->len > 0 && repr->str[repr->len - 1] != ',')
+        g_string_append_c(repr, ',');
+
+      gsize value_len = -1;
+      const gchar *value = filterx_string_get_value(elem, &value_len);
+      str_repr_encode_append(repr, value, value_len, ",");
```

Comment:
I think we need quote escaping as well here.

Reply:
quotes are handled by str_repr_encode()